File: go/day1.go
package main

import (
	"fmt"
	"math"
)

func majority(arr []int) {
	fmt.Printf("Given array: %v", arr)
	count := map[int]int{}
	goal := math.Floor(float64((len(arr) / 2) + 1))
	fmt.Printf("\nGoal: %v appearances", goal)

	for i := range arr {
		if count[arr[i]] != 0 {
			count[arr[i]] += 1
		} else {
			count[arr[i]] = 1
		}
	}

	for num, freq := range count {
		if freq >= int(goal) {
			fmt.Printf("\n%v is the majority element, appearing %v times\n", num, freq)
		}
	}

}

func main() {
	majority([]int{3, 2, 3})
	majority([]int{2, 2, 1, 1, 1, 2, 2})
}

File: java/day1.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class day1 {
    public static void main(String[] arg) {
        ArrayList<Integer> arr1 = new ArrayList<Integer>(Arrays.asList(3, 2, 3));
        ArrayList<Integer> arr2 = new ArrayList<Integer>(Arrays.asList(2, 2, 1, 1, 1, 2, 2));
        majority(arr1);
        majority(arr2);
    }

    public static void majority(ArrayList<Integer> arr) {
        Map<Integer, Integer> count = new HashMap<Integer, Integer>();
        System.out.printf("Given array: %s%n", arr);
        int goal = (int) Math.floor(arr.size()/2)+1;
        System.out.printf("Goal: %s appearances", goal);

        for (int i : arr) {
            if (count.containsKey(i)) {
                count.put(i, count.get(i) + 1);
            } else {
                count.put(i, 1);
            }
        }

        count.forEach(
            (key, value)
            -> {if (value >= goal) {
                System.out.printf("%n%s is the majority element, appearing %s times%n", key, value);
            }}
        );  

    }
}